src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1796,8 +1796,11 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                 if ((flags & DocumentFlags.HasAttachments) == DocumentFlags.HasAttachments)
                     AttachmentsStorage.DeleteAttachmentsOfDocument(context, lowerId, changeVector, modifiedTicks);
 
-                CountersStorage.DeleteCountersForDocument(context, id, collectionName);
-                TimeSeriesStorage.DeleteAllTimeSeriesForDocument(context, id, collectionName);
+                if ((flags & DocumentFlags.HasCounters) == DocumentFlags.HasCounters)
```

Comment:
there is a method for that `.Contain()` IIRC

Reply:
done